### PR TITLE
feat(build): split rendering `index.html` from building `index.json`

### PR DESCRIPTION
**Pull request by fiji-flo**
> [!IMPORTANT]
> **Deprecation notice** 
> Rendering index.html files as part of the build command is now DEPRECATED, and will no longer be supported in Yari v3. To resolve this warning, add the `-n` (`--nohtml`) option.

## Summary

(MP-1174 e.a.)

### Problem

Currently, building the `index.json` files and rendering the `index.html` files happens in a single step via the `yarn build` command. Although it's possible to only build the `index.json` files via `yarn build -n`, it is not possible to only render the `index.html` files.

### Solution

1. Split rendering the `index.html` files into a separate command (`yarn render:html`).
2. Introduce a new explicit command for building the JSON files for docs (`yarn build:docs`).
3. Reuse the `render:html` command to render static pages (aka "SPAs"), by ensuring the corresponding builds produce `index.json`s (that just contain a `pageTitle` and a `url` for now).
4. Show a deprecation warning (see above) when `yarn build` is called **without** the `-n` flag.

---

## Screenshots

(No visual change.)

---

## How did you test this change?

Ran a full build locally on `main` and on this branch, and compared the resulting build output. The only changes identified were as follows:

- There are now `index.json` files for all static pages ("SPAs").
- The `og:url` field now contains the correct URL for static pages ("SPAs"), instead of the base URL.
- All `index.json`s now contain a `url` field, which is replicated in the hydration data of the `index.html`, but not part of the `metadata.json`.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,13 +16,15 @@
...
     "build:blog": "cross-env NODE_ENV=production NODE_OPTIONS='--no-warnings=ExperimentalWarning --loader ts-node/esm' node build/build-blog.ts",
     "build:client": "cd client && cross-env NODE_ENV=production BABEL_ENV=production INLINE_RUNTIME_CHUNK=false node scripts/build.js",
     "build:curriculum": "cross-env NODE_ENV=production NODE_OPTIONS='--no-warnings=ExperimentalWarning --loader ts-node/esm' node build/build-curriculum.ts",
     "build:dist": "tsc -p tsconfig.dist.json",
     "build:glean": "cd client && cross-env VIRTUAL_ENV=venv glean translate src/telemetry/metrics.yaml src/telemetry/pings.yaml -f typescript -o src/telemetry/generated",
+    "build:json": "cross-env NODE_ENV=production NODE_OPTIONS='--no-warnings=ExperimentalWarning --loader ts-node/esm' node build/cli.ts build -n",
     "build:prepare": "yarn build:client && yarn build:ssr && yarn tool popularities && yarn tool spas && yarn tool gather-git-history && yarn tool build-robots-txt",
+    "build:render-html": "cross-env NODE_ENV=production NODE_OPTIONS='--no-warnings=ExperimentalWarning --loader ts-node/esm' node build/cli.ts render",
```
[the remark this answers is not in the document]
Comment (caugner) [reply package.json]: Addressed in fdafc5e7c8, calling `it `render:html` instead.

**Review by caugner — APPROVED** [review]

**Comment by caugner** [comment]
This is now deployed to stage.

**Review by fiji-flo — COMMENTED** [review]
This looks good. Should we get it on stage one more time and then merge?

**Comment by caugner** [comment]
> Should we get it on stage one more time and then merge?

This is still deployed on stage, since 2024-06-14.

**Pull request merged**